scripts/zones/AlTaieu/mobs/Jailer_of_Justice.lua

```diff
@@ -2,29 +2,38 @@
 -- Area: Al'Taieu
 --  NM:  Jailer of Justice
 -----------------------------------
-require("scripts/globals/status");
-require("scripts/globals/magic");
+require("scripts/globals/status")
+require("scripts/globals/magic")
 -----------------------------------
 
+function onMobInitialize(mob)
+    mob:setMobMod(dsp.mobMod.IDLE_DESPAWN, 300)
+end
+
 function onMobFight(mob, target)
-    local popTime = mob:getLocalVar("lastPetPop");
+    local popTime = mob:getLocalVar("lastPetPop")
     -- ffxiclopedia says 30 sec, bgwiki says 1-2 min..
     -- Going with 60 seconds until I see proof of retails timing.
     if (os.time() - popTime > 60) then
-        local alreadyPopped = false;
+        local alreadyPopped = false
         for Xzomit = mob:getID()+1, mob:getID()+6 do
             if (alreadyPopped == true) then
-                break;
+                break
             else
                 if (not GetMobByID(Xzomit):isSpawned()) then
-                    SpawnMob(Xzomit, 300):updateEnmity(target);
-                    mob:setLocalVar("lastPetPop", os.time());
-                    alreadyPopped = true;
+                    SpawnMob(Xzomit, 300):updateEnmity(target)
+                    mob:setLocalVar("lastPetPop", os.time())
+                    alreadyPopped = true
                 end
             end
         end
     end
-end;
+end
+
+function onMobDespawn(mob)
+    for i = 1,6 do DespawnMob(mob:getID() + i) end
```

Comment:
you have duplicate code you already run in OnMobDeath, so I don't think you even need an onMobDespawn at all.